src/uct/ib/base/ib_iface.c

```diff
@@ -354,21 +358,35 @@ void uct_ib_address_pack(const uct_ib_address_pack_params_t *params,
         ib_addr->flags |= UCT_IB_ADDRESS_FLAG_GID_INDEX;
         *(uint8_t*)ptr  = params->gid_index;
     }
+
+    if (params->flags & UCT_IB_ADDRESS_PACK_FLAG_PKEY_VALUE) {
+        ucs_assert(params->pkey_value != UCT_IB_ADDRESS_PKEY_VALUE_DEFAULT);
```

Comment:
maybe just don't pack pkey if it's the default one? this is the logic for subnet prefix for example

Reply:
for subnet_prefix, the equivalent logic is in uct_ib_address_pack, so lr'ts move the default pkey check to uct_ib_address_pack() as well

Reply:
done